### ISSUE TITLE
🐛 False positive DOC503 when Exception is created via class method

### ISSUE THREAD

**Issue by s-weigand** [jsh9/pydoclint#201]
When creating an `Exeption` from a class constructor method a `DOC503` is raised saying `<class name>` does not match `<class name>.<method name>`.

### Example

```py
def ensure_unique_tuple(value: tuple[int, ...]):
    """Ensure that all values in the tuple are unique.

    Parameters
    ----------
    value : tuple[int,...]
        Tuple to check

    Raises
    ------
    TupleDuplicateException
        If the tuple contains duplicate values.
    """
    if len(value) != len(set(value)):
        raise TupleDuplicateException.from_tuple(value)

```
will result in 

```py
DOC503: Function `ensure_unique_tuple` exceptions in the "Raises" section in the docstring do not
match those in the function body Raises values in the docstring: ['TupleDuplicateException'].
Raised exceptions in the body: ['TupleDuplicateException.from_tuple'].
```

### Reproduction

I was preparing a ninja PR to fix this bug (see full reproducible example [on my fork](https://github.com/s-weigand/pydoclint/commit/b725bfa3e3e0d8ac44b8a403c95e6986c6be03a3)).

However, the problem seems to me that at [the point where the `visitor` comes around](https://github.com/jsh9/pydoclint/blob/4021d8d9b3216e2789bd5f28fa7c761228b9b3d4/pydoclint/utils/return_yield_raise.py#L140) there isn't enough information to determine is it of style `module.Exception()` or `Exception.method()` (but might just be my `AST` skill issues 😅).

Here is the `AST` representation of `raise TupleDuplicateException.from_tuple(value)` (thanks [astpretty](https://github.com/asottile/astpretty) to make it more readable 😅)
```py
Raise(
    lineno=45,
    col_offset=8,
    end_lineno=45,
    end_col_offset=55,
    exc=Call(
        lineno=45,
        col_offset=14,
        end_lineno=45,
        end_col_offset=55,
        func=Attribute(
            lineno=45,
            col_offset=14,
            end_lineno=45,
            end_col_offset=48,
            value=Name(lineno=45, col_offset=14, end_lineno=45, end_col_offset=37, id='TupleDuplicateException', ctx=Load()),
            attr='from_tuple',
            ctx=Load(),
        ),
        args=[Name(lineno=45, col_offset=49, end_lineno=45, end_col_offset=54, id='value', ctx=Load())],
        keywords=[],
    ),
    cause=None,
),
```

### Possible solutions
The only thing that comes to my mind currently would be a `remapping` configuration option to run a string replace the output of [`getRaisedExceptions`](https://github.com/jsh9/pydoclint/blob/4021d8d9b3216e2789bd5f28fa7c761228b9b3d4/pydoclint/utils/return_yield_raise.py#L98) (creating an exception from a class method is pretty much an edge case, my actual use case is `ValidationError.from_exception_data` to concatenate validations).

### Env
- `pydoclint==0.5.15`
- py310

**Comment by jsh9** [comment]
Ugh, this is like a whack-a-mole.  Thanks for discovering this edge case.

In https://github.com/jsh9/pydoclint/pull/168/, @Amar1729 added a fix to correctly extracting exception names from things like `a.b.c.d.MyException(msg='my message')`.

But now your use case is the reverse problem.

I traced the code down to this function: https://github.com/jsh9/pydoclint/blob/4021d8d9b3216e2789bd5f28fa7c761228b9b3d4/pydoclint/utils/generic.py#L250-L255

If the `node` is `a.b.c.d.MyException`, `node.value` is `a.b.c.d` and `node.attr` is `MyException`.  But if the `node` is `TupleDuplicateException.from_tuple`, `node.value` is `TupleDpulicateException` and `node.attr` is `from_tuple`.

I haven't been able to find a way effectively distinguish the two.  I wonder whether @Amar1729 has some ideas.

A potential solution is to attempt a partial string match: as long as the docstring exception name is a substring of `node`'s string representation (or maybe a more strict criterion), it's considered a match.  What do you think about this?

**Comment by jsh9** [comment]
My fix is to perform partial string matching instead of strict string matching.  See the code here: https://github.com/jsh9/pydoclint/blob/91263b841745adef4a052354bf4301107fc803ae/pydoclint/utils/generic.py#L250-L268

This way, if you have `a.b.c.d.e.MyException.f.g.h.i.j.k('something')` in your function body, and you have `a.b.c.d.e.MyException` in your docstring, the check passes without any violations.  But if you only have `MyException` in your docstring, _pydoclint_ still shows you a violation --- this is a deliberate design choice which @Amar1729 established.  I'm may be open to discussion on this, if there are compelling reasons to further relax this.

**Comment by Amar1729** [comment]
huh, I'm not sure that I've ever seen that pattern for exceptions before! @jsh9 I think you're right that the only way to distinguish between these two is a substring check like you introduced. I would think the AST alone wouldn't be enough and there would have to be actual name resolution introduced, which seems out of scope.

**Comment by jsh9** [comment]
The fix is published in version 0.5.18.
